From 8fd65be2bfabbc7382d62edfaa27142a600c86ea Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 26 Jul 2021 07:23:41 +0300
Subject: [PATCH] [SW-45031] arch-common: use separate makefile

Restructured makefiles to use separate Makefile
for arch specific common source files.

Change-Id: I596edd5b2986fa968be6569cccf76157fc4b0f62
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/Makefile b/Makefile
index fceaa4a..2ce0406 100644
--- a/Makefile
+++ b/Makefile
@@ -41,13 +41,11 @@
 PREBOOT_LDLIBS = $(LDLIBS) -lfdt
 
 PREBOOT_SRCS_C := $(shell ls lib/common/*.c)
-PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/*.c)
 PREBOOT_SRCS_C += $(shell ls lib/preboot/*.c)
 PREBOOT_SRCS_C += $(shell ls lib/common/lzo/*.c)
 PREBOOT_SRCS_C += $(shell ls app/preboot/*.c)
 PREBOOT_SRCS_C := $(filter-out lib/preboot/thermal_v2.c, $(PREBOOT_SRCS_C))
 PPBOOT_SRCS_C := $(shell ls lib/common/*.c)
-PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/*.c)
 PPBOOT_SRCS_C += $(shell ls app/ppboot/*.c)
 
 include $(HL_DEV_ARCH)/Makefile
diff --git a/app/recovery/Makefile b/app/recovery/Makefile
index 29f0bf4..0700cd7 100644
--- a/app/recovery/Makefile
+++ b/app/recovery/Makefile
@@ -24,7 +24,7 @@
 
 RECOVERY_SRCS_C := $(shell ls *.c)
 RECOVERY_SRCS_C += $(shell ls ../../lib/common/*.c)
-RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/../common/*.c)
+RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/../common/common.c)
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/common.c)
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/$(HL_TARGET).c)
 RECOVERY_APP_S := $(shell ls crt0_recovery.S)
diff --git a/arch/gaudi2/arc/Makefile b/arch/gaudi2/arc/Makefile
index abc7211..e46a262 100644
--- a/arch/gaudi2/arc/Makefile
+++ b/arch/gaudi2/arc/Makefile
@@ -1,3 +1,5 @@
+include $(HL_DEV_ARCH)/../common/Makefile
+
 FW_SIZE := 0x8000
 PPBOOT_MAX_SIZE := 16384
 
diff --git a/arch/gaudi2/arm/Makefile b/arch/gaudi2/arm/Makefile
index c56241b..59568cc 100644
--- a/arch/gaudi2/arm/Makefile
+++ b/arch/gaudi2/arm/Makefile
@@ -1,3 +1,5 @@
+include $(HL_DEV_ARCH)/../common/Makefile
+
 FW_SIZE := 0x4400
 PPBOOT_MAX_SIZE := 20480
 
diff --git a/arch/gaudi2/common/Makefile b/arch/gaudi2/common/Makefile
new file mode 100644
index 0000000..8b002b4
--- /dev/null
+++ b/arch/gaudi2/common/Makefile
@@ -0,0 +1,3 @@
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/common.c)
+
+PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/common.c)
diff --git a/arch/greco/arc/Makefile b/arch/greco/arc/Makefile
index a32aa6f..8f287c2 100644
--- a/arch/greco/arc/Makefile
+++ b/arch/greco/arc/Makefile
@@ -1,3 +1,5 @@
+include $(HL_DEV_ARCH)/../common/Makefile
+
 FW_SIZE := 0x8000
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
diff --git a/arch/greco/arc_elf32/Makefile b/arch/greco/arc_elf32/Makefile
index 3a3cf82..cb7c60f 100644
--- a/arch/greco/arc_elf32/Makefile
+++ b/arch/greco/arc_elf32/Makefile
@@ -1,8 +1,8 @@
+include $(HL_DEV_ARCH)/../common/Makefile
+
 FW_SIZE := 0x8000
 PPBOOT_MAX_SIZE := 16384
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
-PREBOOT_SRCS_C := $(filter-out $(HL_DEV_ARCH)/../common/reset.c, $(PREBOOT_SRCS_C))
-PPBOOT_SRCS_C := $(filter-out $(HL_DEV_ARCH)/../common/reset.c, $(PPBOOT_SRCS_C))
 
 image_validation: image_validation_common
diff --git a/arch/greco/arm/Makefile b/arch/greco/arm/Makefile
index 8316644..e1cded4 100644
--- a/arch/greco/arm/Makefile
+++ b/arch/greco/arm/Makefile
@@ -1,3 +1,5 @@
+include $(HL_DEV_ARCH)/../common/Makefile
+
 FW_SIZE := 0x4400
 PPBOOT_MAX_SIZE := 18432
 PREBOOT_SRCS_C += $(shell ls lib/preboot/thermal_v2.c)
diff --git a/arch/greco/common/Makefile b/arch/greco/common/Makefile
new file mode 100644
index 0000000..c693c15
--- /dev/null
+++ b/arch/greco/common/Makefile
@@ -0,0 +1,4 @@
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/common.c)
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/reset.c)
+
+PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/../common/common.c)
